### DISCUSSION TITLE
Stored Batches ?

### DISCUSSION THREAD

**Discussion by Mr-Dispatch** [orientechnologies/orientdb#9969] [Q&A] [answered]
Batches are, as has been pointed-out to me, a power-house. They are basically stored procedures, without the stored part.

Allowing for them being stored allows (besides better staging for changes, and keeping the same interface / code for clients) for some further niceties, such as impersonation / information hiding.
Oracle et al. allow for stored procedure to access objects (tables, views, grants, whatever) with the rights of the creator/owner, without the user having to have the explicit rights delegated. This is nice to have, even in addition to table/column/row/whatever ACL's.

**Comment by Mr-Dispatch** [comment]
Other possibilities are storing a more efficient / pre-parsed form, caching some optimal representation on invocation, ...???...

**Comment by tglman** [comment, marked as answer]
Hi,

OrientDB has Functions that do this exact job, here is how to create one with sql: http://orientdb.com/docs/3.2.x/sql/SQL-Create-Function.html

Bye
